language/move-lang/src/parser/syntax.rs

```diff
@@ -874,6 +882,43 @@ fn parse_term(context: &mut Context) -> Result<Exp, Diagnostic> {
             Exp_::Continue
         }
 
+        Tok::IdentifierValue if context.tokens.content() == VECTOR_IDENT => {
```

Comment:
@bob-wilson thoughts on this hack?

Particularly on the casing of the next token being `<` or `[`. 
Should we just make `vector` a conditional keyword? i.e. a keyword when in an exp context? It is already restricted from being used as a name for anything (anything other than the type) so it won't change much

Reply:
I don't know what "keyword when in an exp context" would mean. The lexer doesn't (currently) distinguish the context, so we would need to add that extra complexity. Maybe that would still be cleaner and/or more maintainable this this hack? From the user's point of view, would it change anything? I don't have much of an opinion offhand. Maybe @modocache has something to say about the potential impact on regexps for syntax coloring?

Reply:
There is no extra complexity. We already conditional keyword all over the place, you just case out the content of the identifier before allowing the parser to continue. Which is basically what has happened here except it reverts back to a name if the next token isn't `<` or `\s*[`

Reply:
@bob-wilson I added a fixup that deals just with this issue. With the fixup, `vector` is _not_ a conditional keyword in expressions. This has a bunch of downstream effects you can see in tests

Reply:
This version is simpler and disallowing "vector" for variable/module names seems like a good idea anyway to avoid confusion, so this looks good to me.